.github/actions/test-package/action.yml

```diff
@@ -10,43 +11,64 @@ inputs:
   llvm:
     description: 'LLVM version to use.'
     required: true
+  dockerfile:
+    description: 'Hardcode the path of the dockerfile to use.'
+    required: false
+    default: .github/workflows/Dockerfile
+  build-package:
+    description: 'Script which builds the given package.'
+    required: true
+  test-package:
+    description: 'Script which tests the given package.'
+    required: true
+  pkg-name:
+    description: 'Where to move the package.'
+    required: false
+    default: package.pkg
+
 runs:
   using: 'composite'
+
   steps:
+
   - name: 'Check out code'
     uses: actions/checkout@v3
     with:
       path: k-${{ inputs.distro }}
       submodules: recursive
+
   - name: 'Set up Docker'
     uses: ./.github/actions/with-docker
     with:
-      tag: k-package-build-${{ inputs.distro }}-${{ github.sha }}
+      tag: k-package-build-${{ inputs.os }}-${{ inputs.distro }}-${{ github.sha }}
       subdir: k-${{ inputs.distro }}/
       os: ${{ inputs.os }}
       distro: ${{ inputs.distro }}
       llvm: ${{ inputs.llvm }}
+      dockerfile: ${{ inputs.dockerfile }}
 
   - name: 'Build Package: ${{ inputs.distro }}'
     shell: bash {0}
     env:
       BASE_DISTRO: ${{ inputs.distro }}
+      BASE_OS: ${{ inputs.os }}
+      BUILD_PACKAGE: ${{ inputs.build-package }}
+      PKG_NAME: ${{ inputs.pkg-name }}
     run: |
       set -euxo pipefail
-      docker exec -t k-package-build-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c 'mv package/debian debian'
-      docker exec -t k-package-build-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c "mv debian/compat.${BASE_DISTRO} debian/compat"
-      docker exec -t k-package-build-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c "mv debian/control.${BASE_DISTRO} debian/control"
-      docker exec -t k-package-build-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c "mv debian/rules.${BASE_DISTRO} debian/rules"
-      docker exec -t k-package-build-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c 'dpkg-buildpackage'
+      docker exec -t k-package-build-${BASE_OS}-${BASE_DISTRO}-${GITHUB_SHA} /bin/bash -c "${BUILD_PACKAGE} ${PKG_NAME}"
```

Comment:
Here we call the `BUILD_PACKAGE` script, which we assume moves the created package to `PKG_NAME` location.